core/controllers/skill_editor_test.py

```diff
@@ -52,41 +79,75 @@ def test_access_skill_editor_page(self):
-                expect_errors=True)
+                self.url, expect_errors=True)
             self.assertEqual(response.status_int, 401)
             self.logout()
 
             # Check that admins can access and edit in the editor page.
             self.login(self.ADMIN_EMAIL)
-            response = self.testapp.get(
-                '%s/%s' % (feconf.SKILL_EDITOR_URL_PREFIX, self.skill_id))
+            response = self.testapp.get(self.url)
             self.assertEqual(response.status_int, 200)
             self.logout()
 
-    def test_editable_skill_handler_get(self):
-        # Check that non-admins cannot access the editable skill data.
-        with self.swap(feconf, 'ENABLE_NEW_STRUCTURES', True):
-            self.login(self.NEW_USER_EMAIL)
-            response = self.testapp.get(
-                '%s/%s' % (
-                    feconf.SKILL_EDITOR_DATA_URL_PREFIX, self.skill_id),
-                expect_errors=True)
-            self.assertEqual(response.status_int, 401)
-            self.logout()
+    def test_skill_editor_page_fails(self):
+        self.login(self.ADMIN_EMAIL)
+        # Check GET returns 404 when new strutures' pages are not enabled.
+        with self.swap(feconf, 'ENABLE_NEW_STRUCTURES', False):
+            response = self.testapp.get(self.url, expect_errors=True)
+            self.assertEqual(response.status_int, 404)
 
-            # Check that admins can access the editable skill data.
+        # Check GET returns 404 when cannot get skill by id.
+        feconf_swap = self.swap(feconf, 'ENABLE_NEW_STRUCTURES', True)
+        skill_services_swap = self.swap(
+            skill_services, 'get_skill_by_id', self._mock_get_skill_by_id)
+        with feconf_swap, skill_services_swap:
             self.login(self.ADMIN_EMAIL)
-            json_response = self.get_json(
-                '%s/%s' % (
-                    feconf.SKILL_EDITOR_DATA_URL_PREFIX, self.skill_id))
-            self.assertEqual(self.skill_id, json_response['skill']['id'])
-            self.logout()
+            response = self.testapp.get(self.url, expect_errors=True)
+            self.assertEqual(response.status_int, 404)
+        self.logout()
+
 
-    def test_editable_skill_handler_put(self):
-        # Check that admins can edit a skill.
-        change_cmd = {
+class SkillRightsHandlerTest(BaseSkillEditorControllerTest):
+    """Tests for SkillRightsHandler."""
+
+    def setUp(self):
+        super(SkillRightsHandlerTest, self).setUp()
+        self.url = '%s/%s' % (feconf.SKILL_RIGHTS_URL_PREFIX, self.skill_id)
+
+    def test_skill_rights_handler_succeeds(self):
+        self.login(self.ADMIN_EMAIL)
+        # Check that admins can access and edit in the editor page.
+        self.get_json(self.url)
+        # Check GET returns JSON object with can_edit_skill_description set
+        # to False if the user is not allowed to edit the skill description.
+        def _get_all_actions(*_args):
```

Comment:
_mock_get_all_actions, and make into class-level function (like in BaseSkillEditorControllerTest). Add short docstring explaining what it's doing conceptually.

Even better: perhaps just test the permissions directly (without a mock) for non-admin users.